pkg/instance/instance.go

```diff
@@ -0,0 +1,588 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	spdkapi "github.com/longhorn/longhorn-spdk-engine/pkg/api"
+	spdkclient "github.com/longhorn/longhorn-spdk-engine/pkg/client"
+
+	"github.com/longhorn/longhorn-instance-manager/pkg/client"
+	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+	"github.com/longhorn/longhorn-instance-manager/pkg/meta"
+	"github.com/longhorn/longhorn-instance-manager/pkg/types"
+)
+
+type Server struct {
+	logsDir                      string
+	processManagerServiceAddress string
+	diskServiceAddress           string
+	spdkServiceAddress           string
+	spdkEnabled                  bool
+	shutdownCh                   chan error
+	HealthChecker                HealthChecker
+}
+
+func NewServer(logsDir, processManagerServiceAddress, diskServiceAddress, spdkServiceAddress string, spdkEnabled bool, shutdownCh chan error) (*Server, error) {
+	s := &Server{
+		logsDir:                      logsDir,
+		processManagerServiceAddress: processManagerServiceAddress,
+		diskServiceAddress:           diskServiceAddress,
+		spdkServiceAddress:           spdkServiceAddress,
+		spdkEnabled:                  spdkEnabled,
+		shutdownCh:                   shutdownCh,
+		HealthChecker:                &GRPCHealthChecker{},
+	}
+
+	go s.startMonitoring()
+
+	return s, nil
+}
+
+func (s *Server) startMonitoring() {
+	done := false
+	for {
+		select {
+		case <-s.shutdownCh:
+			logrus.Info("Instance Server is shutting down")
+			done = true
```

Comment:
nit: done is not necessary, can just use break.

Reply:
Removed. Thank you.